apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -884,11 +886,28 @@ defmodule Astarte.RealmManagement.Queries do
   def get_jwt_public_key_pem(realm_name) do
     keyspace = Realm.keyspace_name(realm_name)
 
-    KvStore.fetch_value("auth", "jwt_public_key_pem", :string,
-      prefix: keyspace,
-      consistency: :quorum,
-      error: :public_key_not_found
-    )
+    try do
+      KvStore.fetch_value("auth", "jwt_public_key_pem", :string,
+        prefix: keyspace,
+        consistency: :quorum,
+        error: :public_key_not_found
+      )
+    rescue
+      error in Xandra.Error ->
+        %Xandra.Error{message: message} = error
+
+        case Regex.run(@keyspace_does_not_exist_regex, message) do
+          [_message, keyspace] ->
+            Logger.warning("Keyspace #{keyspace} does not exist.",
+              tag: "realm_not_found"
+            )
+
+            {:error, :realm_not_found}
+
+          nil ->
+            raise error
+        end
+    end
```

Comment:
Why can't this one explode like the rest of the queries?
The exception should be handled by RPC.